vignettes/evaluate-tree-registration.qmd

```diff
@@ -0,0 +1,142 @@
+---
+title: "Evaluating tree registration"
+author: "David Russell"
+format: html
+editor: visual
+---
+
+## Purpose
+
+There are various approaches to tree registration. The goal of this notebook is to conduct a direct comparison between them. The user can control the types of errors present in the initial registration and define a function that evaluates the quality of the predicted registration. Then, this code runs a set of experiments where a simulated tree alignment problem is generated, each algorithm is run on it, and the results are scored.
+
+## Setup
+
+Load the the `ofo` package (which is the repo that this vignette's source file is in, so we can use `devtools::load_all()` to load all the package's functions).
+
+```{r}
+library(devtools)
+devtools::load_all()
+```
+
+## Define the registration algorithms to compare
+
+```{r}
+# The actual functions that take in a predicted and observed tree map and output a transform between them
+REGISTRATION_METHODS = c(find_best_shift, find_best_shift)
+# Additional arguments that are provided to these functions to control their behavior
+PER_METHOD_REGISTRATION_ARGUMENTS = list(list(method="optim"), list(method="grid"))
+# The name of these methods for visualization
+REGISTRATION_METHOD_NAMES = c("find_best_shift_optim", "find_best_shift_grid")
+```
+
+# Define the experiments to run
+
+The values will be passed to `simulate_tree_maps` to generate experiments to run. You can provide a single value or a sequence of values. If you provide a sequence, all combinations of values can be run across the different arguments. This number can get large, so you choose to select only a random sample of them.
+
+```{r}
+# All other values will be set to the default from simulate_tree_maps
+MAP_PARAMS = list(
+  shift_x=seq(from=-20, to=20),
+  shift_y=as.vector(1),
+  horiz_jitter = seq(from=0, to=10, by=1)
+)
+# Choose this many random samples from the full list of combinations. Set to NULL to use all combinations.
+N_RANDOM_SAMPLES = 20
+```
+
+## Run experiments
+
+Conduct all the registration trials with the set of starting configurations and multiple algorithms.
+
+```{r}
+all_registration_results = test_tree_registration(
+ map_params = MAP_PARAMS,
+ registration_methods = REGISTRATION_METHODS,
+ registration_method_names = REGISTRATION_METHOD_NAMES,
+ per_method_registration_arguments = PER_METHOD_REGISTRATION_ARGUMENTS,
+ n_random_samples = N_RANDOM_SAMPLES
+)
+```
+
+## Define evaluation functions
+
+These are used only to visualize the results. They take the true and predicted registration values and use them to compute a scalar metric representing the quality.
+
+```{r}
+# Return the euclidian distance between the predicted and true shift
+eval_res_dist = function(predicted_shift, map_config){
+  pred_shift_vec = as.numeric(list(x=predicted_shift$shift_x, y=predicted_shift$shift_y))
```

Comment:
I think you can simplify this as:

`pred_shift_vec = c(predicted_shift$shift_x, predicted_shift$shift_y)`,

optionally prefixing the arguments of `c` with `x=` and `y=`, but your `as.numeric` call removes those names anyway.

Reply:
It seems maybe you were setting up to accept a `predicted_shift` parameter with multiple rows of predicted shifts, but I don't think this code exactly accommodates that, nor does that situation arise in the current implementation.